playbooks/roles/katello_provisioning/tasks/katello.yml

```diff
@@ -0,0 +1,92 @@
+  register: katello_provisioning_product
+  ignore_errors: True
+
+- name: 'create centos product'
+  shell: >
+    {{ katello_provisioning_hammer }} product create
+    --name "CentOS"
+    --organization "{{ katello_provisioning_organization }}"
+  when: katello_provisioning_product.stderr.find('not found') != -1
+
+# Repo
+- name: 'find repo'
+  shell: >
+    {{ katello_provisioning_hammer }} repository info --name "CentOS 7" --product "CentOS" --organization "{{ katello_provisioning_organization }}"
+  register: katello_provisioning_repo
+  ignore_errors: True
+
+- name: 'create centos repo'
+  shell: >
+    {{ katello_provisioning_hammer }} repository create
+    --product="CentOS"
+    --content-type="yum"
+    --name "CentOS 7"
+    --url http://mirror.centos.org/centos/7/os/x86_64
+    --download-policy on_demand
+    --organization "{{ katello_provisioning_organization }}"
+  when: katello_provisioning_repo.stderr.find('not found') != -1
+
+- name: 'sync the repo'
+  shell: >
+    {{ katello_provisioning_hammer }} repository synchronize
+    --name "CentOS 7"
+    --product "CentOS"
+    --organization "{{ katello_provisioning_organization }}"
+
+# Activation key
+- name: 'find activation key'
+  shell: >
+    {{ katello_provisioning_hammer }} activation-key info --name "CentOS 7" --organization "{{ katello_provisioning_organization }}"
+  register: katello_provisioning_activation_key
+  ignore_errors: True
+
+- name: 'create activation key'
+  shell: >
+    {{ katello_provisioning_hammer }} activation-key create
+    --organization "{{ katello_provisioning_organization }}"
+    --name="CentOS 7"
+    --content-view="Default Organization View"
+    --lifecycle-environment="Library"
+    --unlimited-hosts
+  when: katello_provisioning_activation_key.stderr.find('not found') != -1
+
+- name: 'medium env should be in locs'
+  shell: >
+    {{ katello_provisioning_hammer }} medium update
+    --name "{{ katello_provisioning_organization|regex_replace('\s', '_') }}/Library/CentOS/CentOS_7"
+    --organizations "{{ katello_provisioning_organization }}"
+    --locations "{{ katello_provisioning_location }}"
+
+# Host group
+- name: 'find host group'
+  shell: >
+    {{ katello_provisioning_hammer }} hostgroup info --name "Forklift CentOS 7"
+  register: katello_provisioning_hg
+  ignore_errors: True
+
+- name: 'create a host group'
+  shell: >
+    {{ katello_provisioning_hammer }} hostgroup create
+    --name "Forklift CentOS 7"
+    --architecture x86_64
+    --content-view "Default Organization View"
+    --domain example.com
+    --lifecycle-environment Library
+    --operatingsystem "CentOS 7"
+    --root-pass changeme
+    --organizations "{{ katello_provisioning_organization }}"
+    --query-organization "{{ katello_provisioning_organization }}"
+    --partition-table "Kickstart default"
+    --medium "{{ katello_provisioning_organization|regex_replace('\s', '_') }}/Library/CentOS/CentOS_7"
```

Comment:
You could make this into a variable so you are performing this operation on the data in only one spot